src/native/fs/fs_napi.cpp

```diff
@@ -991,6 +995,38 @@ struct FileSetxattr : public FSWrapWorker<FileWrap>
     }
 };
 
+struct linkFileAt : public FSWrapWorker<FileWrap>
+{
+    std::string _path;
+    std::string _filepath;
+    int _dirfd;
+    int _rc_res;
+
+
+    linkFileAt(const Napi::CallbackInfo& info)
+        : FSWrapWorker<FileWrap>(info)
+    {
+        _path = info[1].As<Napi::String>();
+        _filepath = info[2].As<Napi::String>();
+        _dirfd = _wrap->_fd;
+    }
+    virtual void Work()
+    {
+        _rc_res = gpfs_linkat(_dirfd, "", AT_FDCWD, _filepath.c_str(), AT_EMPTY_PATH);
+        if (_rc_res) {
+            DBG0("linkFileAt error FS::FD " << DVAL(_dirfd) << DVAL(_rc_res));
+            SetSyscallError();
+            return;
+        }
+    }
+    virtual void OnOK()
+    {
+        DBG1("IBM::linkFileAt::OnOK: " << DVAL(_rc_res));
+        Napi::Env env = Env();
+        _deferred.Resolve(Napi::Number::New(env, _rc_res));
+    }
```

Comment:
I think you should omit this entire function. We don't have anything to return as _rc_res was already checked and is 0...